app/cdap/components/PostRunActions/WizardNav/index.tsx

```diff
@@ -0,0 +1,272 @@
+import { MyArtifactApi } from 'api/artifact';
+import { WizardConfigureConfirmStep } from './WizardConfigureConfirmStep';
+import PipelineConfigurationsStore, {
+  ACTIONS as PipelineConfigurationsActions,
+} from 'components/PipelineConfigurations/Store';
+
+export interface IWizardNavProps {
+  mode: string;
+  artifact: any;
+  setAction: (action: any) => void;
+  actionCreator: any;
+  toggle: () => void;
+  getPostActions: () => void;
+  action: object;
+  setShowValidateButton: (show: boolean) => void;
+  propertyErrors: object;
+  validatePluginProperties: (action: any, errorCb: any) => void;
+  setState: (func: any) => void;
+}
+
+function findLatestArtifact(plugin) {
+  const allArtifacts = plugin.allArtifacts;
+  const versions = allArtifacts.map((pa) => pa.artifact.version);
+  const latestVersion = findHighestVersion(versions, true);
+  return allArtifacts.find((pa) => pa.artifact.version === latestVersion);
+}
+
+// TODO: sumengwang clean up the necessary properties
+function filterPlugins(results) {
+  const pluginsMap = {};
+  results.forEach((plugin) => {
+    if (!pluginsMap[plugin.name]) {
+      pluginsMap[plugin.name] = Object.assign({}, plugin, {
+        defaultArtifact: plugin.artifact,
+        allArtifacts: [],
+      });
+    }
+    pluginsMap[plugin.name].allArtifacts.push(plugin);
+    const latestArtifact = findLatestArtifact(pluginsMap[plugin.name]);
+    pluginsMap[plugin.name].defaultArtifact = latestArtifact.artifact;
+    pluginsMap[plugin.name].plugin = {};
+    pluginsMap[plugin.name].plugin.name = plugin.name;
+    pluginsMap[plugin.name].plugin.artifact = latestArtifact.artifact;
+    pluginsMap[plugin.name].plugin.type = plugin.type;
+    pluginsMap[plugin.name].plugin.properties = {};
+    pluginsMap[plugin.name].properties = {};
+  });
+  return pluginsMap;
+}
+
+export const WizardNav = (props: IWizardNavProps) => {
+  const initialState = {
+    currentStage: 1,
+    selectedAction: null,
+    widgetJson: null,
+    values: null,
+  };
+
+  function pluginFetch(action) {
+    if (action) {
+      const { name, version, scope } = action.plugin.artifact;
+      const params = {
+        namespace: getCurrentNamespace(),
+        artifactId: name,
+        version,
+        scope,
+        extensionType: action.plugin.type,
+        pluginName: action.plugin.name,
+      };
+
+      MyArtifactApi.fetchPluginDetails(params).subscribe((res) => {
+        action._backendProperties = res[0].properties;
+        fetchWidgets(action);
+      });
+    }
+  }
+
+  // Fetching Widget JSON for the plugin
+  function fetchWidgets(action) {
+    const { name, version, scope } = action.plugin.artifact;
+    const widgetKey = `widgets.${action.plugin.name}-${action.plugin.type}`;
+    const widgetParams = {
+      namespace: getCurrentNamespace(),
+      artifactName: name,
+      scope,
+      artifactVersion: version,
+      keys: widgetKey,
+    };
+    MyPipelineApi.fetchWidgetJson(widgetParams).subscribe((res) => {
```

Comment:
@njbriggs we should look into making api creator into a custom hook generator

Reply:
👍